src/libmongoc/src/mongoc/mongoc-host-list.c

```diff
@@ -236,15 +236,31 @@ _mongoc_host_list_from_string_with_err (mongoc_host_list_t *link_,
       /* if present, the port should immediately follow after ] */
       sport = strchr (close_bracket, ':');
       if (sport > close_bracket + 1) {
+         bson_set_error (error,
+                         MONGOC_ERROR_COMMAND,
+                         MONGOC_ERROR_COMMAND_INVALID_ARG,
+                         "If present, port should immediately follow the \"]\""
+                         "in an IPv6 address"
+                         );
```

Comment:
Nice, I agree with using `MONGOC_ERROR_COMMAND` and `MONGOC_ERROR_COMMAND_INVALID_ARG` as the domain and code. This would error if the user supplied a wrong IPv6 address in the URI, or if a server were to report an invalid IPv6 address.

I looked at codecov.io and it looks like this case isn't encountered in tests.
https://codecov.io/gh/mongodb/mongo-c-driver/src/master/src/libmongoc/src/mongoc/mongoc-host-list.c
`test-mongoc-uri.c` has `test_mongoc_host_list_from_string` which may be an easy way to add a test for this case. Adding the test is out of the scope of this ticket, but if it's a quick win I suggest that we add it.

As a sidenote, codecov.io is not very accurate. It does not capture tests that run on platforms other than Linux, and it fails to upload very often. But it's still a useful measure sometimes.